torch/_dynamo/symbolic_convert.py

```diff
@@ -1909,70 +1909,73 @@ def __init__(
             f_code=f_code,
             export=export,
         )
-        self.one_graph: bool = one_graph
-        self.export = export
-        self.mutated_closure_cell_contents = mutated_closure_cell_contents
-        if self.export:
-            assert (
-                self.one_graph
-            ), "Export without one graph - something has gone wrong."
-
-        vars = list(code_options["co_varnames"])
-        vars.extend(x for x in self.cell_and_freevars() if x not in vars)
-
-        self.symbolic_locals = collections.OrderedDict(
-            (
-                k,
-                VariableBuilder(
-                    self,
-                    LocalSource(k)
-                    if k in code_options["co_varnames"]
-                    else LocalSource((k)),
-                )(f_locals[k]),
+        # as soon as we create the tracing context we should keep it active, so any calls
+        # into dynamo apis can rely on finding it
+        with tracing(self.output.tracing_context):
```

Comment:
@voznesenskym I think it might actually be a cleaner change if I could activate TracingContext outside the InstructionTranslator init call in convert_frame.py.

However, to do that i'd have to move the FakeMode and ShapeEnv constructors also out to convert_frame.py.  I am not actually sure if that's a bad change, lmk if you prefer that way or this PR.

Reply:
Just trying to understand - Why this is better?

Reply:
Updated pr desc to clarify, lmk if still not clear.